File: src/main/java/oscar/visual/MapLine.java
/*******************************************************************************
 * This file is part of OscaR (Scala in OR).
 *   
 * OscaR is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2.1 of the License, or
 * (at your option) any later version.
 *  
 * OscaR is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *  
 * You should have received a copy of the GNU General Public License along with OscaR.
 * If not, see http://www.gnu.org/licenses/gpl-3.0.html
 ******************************************************************************/
package oscar.visual;


/**
 * @author Pierre Schaus
 */
public class MapLine {
	double lt1, lg1, lt2, lg2;
	Map map;

	public MapLine(Map map,double lt1, double lg1, double lt2, double lg2) {
		this.map = map;
		this.lt1 = lt1;
		this.lg1 = lg1;
		this.lt2 = lt2;
		this.lg2 = lg2;
	}

	public void setDest(double lt2, double lg2) {
		this.lt2 = lt2;
		this.lg2 = lg2;
		map.viewer.repaint();

	}

	public void setOrig(double lt1, double lg1) {
		this.lt1 = lt1;
		this.lg1 = lg1;
		map.viewer.repaint();
	}		
}

File: oscar-visual/src/main/scala/oscar/visual/map/MapLine.scala
/*******************************************************************************
 * OscaR is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 2.1 of the License, or
 * (at your option) any later version.
 *   
 * OscaR is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License  for more details.
 *   
 * You should have received a copy of the GNU Lesser General Public License along with OscaR.
 * If not, see http://www.gnu.org/licenses/lgpl-3.0.en.html
 ******************************************************************************/
package oscar.visual.map

import java.awt.Color


/**
 * @author Pierre Schaus
 */
class MapLine(map : VisualMap,  lat1 : Double, long1 : Double, lat2 : Double, long2 : Double, col : Color = Color.RED) {
	private var olat = lat1
	private var olong = long1
	private var dlat = lat2
	private var dlong = long2
	
	var color = col

	def this(map : VisualMap, o: (Double, Double), d: (Double, Double), col : Color = Color.RED) = this(map, o._1, o._2, d._1, d._2, col)
	def remove = map.removeLine(this)
	
	def dest_=(d:(Double,Double)) : Unit = {
		this.dlat = d._1;
		this.dlong = d._2;
		map.viewer.repaint();
	}

	def orig_=(o:(Double,Double) ) : Unit =  {
		this.olat = o._1;
		this.olong = o._2;
		map.viewer.repaint();
	}		
	
	def dest = (dlat, dlong)
	def orig = (olat, olong)
}
